src/client/pages/Section/Descriptions/NationalDataDescriptions/DataSources/Columns/YearForDataSource.tsx

```diff
@@ -37,10 +38,16 @@ const YearForDataSource: React.FC<Props> = (props: Props) => {
       data-tooltip-content={validationError ? t('generalValidation.notEmpty') : ''}
       data-tooltip-id={TooltipId.error}
       editable={!disabled}
+      highlighted={highlighted}
```

Comment:
Those components are wrapping the DataCell, so the `highlighted` prop is passed to those wrappers but not the DataCell component itself. 
Couple of ideas to avoid defining the prop every time:
1. Pass the prop like this so it's not necessary to define it:
```
<DataCell
      className={classNames({ 'validation-error': validationError })}
      data-tooltip-id={TooltipId.error}
      ...
      // eslint-disable-next-line react/jsx-props-no-spreading
      {...props}
    >
```
2. Use context api in DataRow and DataCell (maybe not the best for performance)
3. In DataRow, check recursively if any child component (or child of a child) is a DataCell, and pass the props directly to it (also maybe not the best for performance).
If you can think of any other please let me know.

I think defining and passing the `highlighted` prop makes the logic easier to follow. It's like we already do with other props like `lastCol`, `lastRow` and `disabled`.

Reply:
We are wrapping the DataCells with DataRow, yes. The problem is the data sources grid components also wrap the DataCell and capture the added props from DataRow:
normally it would be:
```
DataRow -> adds highlighted prop
   DataCell -> gets the highlighted prop
```
but with the Data Source component also wrapping the DataCell:
```
DataRow -> adds highlighted prop
   YearForDataSource -> gets the highlighted prop
      DataCell -> renders with props passed in YearForDataSource component
```
To avoid that in-between component from capturing the props from DataRow we would need to implement one of those 3 ideas I mentioned before.

Reply:
Added it in the latest commit👍 
To avoid the prop drilling, the DataCells need to be direct children of DataRow, so I updated the `DataSourceRow` component accordingly.

Reply:
The only place left to change I see is the `ContactRow` component. That is a bit of an special case because there we use the generic component `CellNodeExt`, which returns a `DataCell` :
```
ContactRow
  DataRow 
    CellNodeExt -> gets the extra props from DataRow and does not pass them unless prop drilling
       DataCell 
         Component -> a component based  on column type (multiselect, text, or link, etc)
```
Since `CellNodeExt` is a generic component, do we want to add there the optional props `highlighted` and `firstCol`?

Reply:
Since `CellNodeExt` is only used in two places, yes. I used the same approach to keep it consistent 👍 
![image](https://github.com/openforis/fra-platform/assets/41337901/6fb839b0-1cdc-46d2-8e85-6fb767e52e00)